akd/src/proof_structs.rs

```diff
@@ -33,9 +33,22 @@ pub struct MembershipProof<H: Hasher> {
     pub dirs: Vec<Direction>,
 }
 
+// Manual implementation of Clone, see: https://github.com/rust-lang/rust/issues/41481
+impl<H: Hasher> Clone for MembershipProof<H> {
+    fn clone(&self) -> Self {
+        Self {
+            label: self.label,
+            hash_val: self.hash_val,
```

Comment:
```Hasher``` implements copy natively?

Reply:
Not `Hasher`, but `Hasher::Digest` does implement copy